src/gdext/surface/pragmas.nim

```diff
@@ -0,0 +1,33 @@
+import gdext/core/userclass/propertyinfo
+
+type ExpEasingArgument* = enum
+  attenuation, positive_only
+type RangeArgument*  = enum
+  or_less, or_greater, exp, radians_as_degrees, degrees, hide_slider
+
+template gdexport*() {.pragma.}
+template gdexport_multiline*() {.pragma.}
+template gdexport_color_no_alpha*() {.pragma.}
+template gdexport_dir*() {.pragma.}
+template gdexport_global_dir*() {.pragma.}
+template gdexport_file*() {.pragma.}
+template gdexport_global_file*() {.pragma.}
+template gdexport_flags_2d_navigation*() {.pragma.}
+template gdexport_flags_2d_physics*() {.pragma.}
+template gdexport_flags_2d_render*() {.pragma.}
+template gdexport_flags_3d_navigation*() {.pragma.}
+template gdexport_flags_3d_physics*() {.pragma.}
+template gdexport_flags_3d_render*() {.pragma.}
+template gdexport_flags_avoidance*() {.pragma.}
+template gdexport_storage*() {.pragma.}
+template gdexport_exp_easing*() {.pragma.}
+
+template gdexport_enum*(cases: varargs[string]) {.pragma.}
+template gdexport_flags*(flags: varargs[string]) {.pragma.}
+template gdexport_exp_easing*(extra: ExpEasingArgument) {.pragma.}
+
+template gdexport_range*(min, max: SomeNumeric, extra: varargs[RangeArgument]) {.pragma.}
+template gdexport_range*(min, max, step: SomeNumeric, extra: varargs[RangeArgument]) {.pragma.}
```

Comment:
I initially created a macro that will generate these as well, but I thought that might be going a bit overboard with the "magic" :)

I thought you can chime in as well - generating these would be an additional complexity layer but could potentially simplify maintenance if Godot adds new export types. Then again, how often does that happen? Probably better to leave as-is :)

<details>
<summary>The code for generating</summary>

```nim
var ignoredParams {.compileTime.}: seq[string] = @["static[alias]", $SomeInt, $SomeColor, $SomeString, $SomeProperty, $SomeUserClass]

macro genPragma*(exportMacro: untyped) =
    let name = exportMacro[0][1]
    let params = exportMacro.findChild(it.kind == nnkFormalParams)
    var templateParams: seq[NimNode] = @[newNimNode(nnkEmpty)]

    for param in params:
        if param.kind != nnkIdentDefs:
            continue
   
        let paramTypes: seq[NimNode] = case param[1].kind:
            of nnkEmpty:
                error("Untyped args not supported")
                @[]
            of nnkInfix:
                param[1][1..^1]
            else:
                @[param[1]]
        
        var containsIgnoredParams = false
        for paramType in paramTypes:
            if ignoredParams.contains $paramType.toStrLit:
                containsIgnoredParams = true
        
        if not containsIgnoredParams:
            templateParams.add(param)
    
    let metaPragma = newNimNode(nnkPragma).add(newIdentNode("pragma"))
    let templ = newNimNode(nnkTemplateDef).add(
        name,
        newEmptyNode(),
        newEmptyNode(),
        newNimNode(nnkFormalParams).add(templateParams),
        metaPragma,
        newEmptyNode(),
        newEmptyNode())

    result = newStmtList()
    result.add exportMacro
    result.add templ
```

Usage: 
```nim
macro gdexport*(iden: SomeProperty, alias: static[alias] = noAlias) {.genPragma.}  =
   ...
```

Reply:
I agree with you.
In addition, I can say, as I have made this mistake many times in this project, that macro is quite heavy. If you use it too much where it is not to the benefit of the user, the compile time will only increase.